### PR TITLE
Run `make drone` to use the new build image

### PR DESCRIPTION
**Pull request by dannykopping**
**What this PR does / why we need it**:
Updates the `.drone/.drone.yml` file after #5250 was merged.

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Special notes for your reviewer**:

**Checklist**
- [ ] Documentation added
- [ ] Tests updated
- [ ] Add an entry in the `CHANGELOG.md` about the changes.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -42,6 +42,8 @@ DOCKER_IMAGE_DIRS := $(patsubst %/Dockerfile,%,$(DOCKERFILES))
 # make BUILD_IN_CONTAINER=false target
 # or you can override this with an environment variable
 BUILD_IN_CONTAINER ?= true
+
+# ensure you run `make drone` after changing this
 BUILD_IMAGE_VERSION := 0.19.0
```
Comment (jeschkies) [inline Makefile:45]: That should be 0.20.0 now. Thanks for taking this over.

**Review by jeschkies — CHANGES_REQUESTED** [review]
🙏

**Comment by jeschkies** [comment]
@dannykopping could you pull in my changes from https://github.com/jeschkies/loki/tree/danny/make-drone?

**Comment by dannykopping** [comment]
> @dannykopping could you pull in my changes from https://github.com/jeschkies/loki/tree/danny/make-drone?

Done, and rebased against `main`

File: .drone/drone.yml [hunk 2]
```diff
@@ -1,73 +1,40 @@
 ---
 kind: pipeline
-name: loki-build-image
```
Comment (jeschkies) [inline .drone/drone.yml:3]: Why is this removed?

**Review by jeschkies — APPROVED** [review]

**Pull request merged**